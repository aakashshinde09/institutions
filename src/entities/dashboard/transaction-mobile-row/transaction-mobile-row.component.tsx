import type { TransactionLookup } from '@api/data-contracts'
import { trxType } from '@shared/lib/constants'
import { formatDate, formatNumber, shortenWallet } from '@shared/lib/utils'
import { CaptionItem } from '@shared/ui/caption-item'

interface ITransactionMobileRowProperties {
  data: TransactionLookup
}

export const TransactionMobileRow = ({ data }: ITransactionMobileRowProperties) => {  
  return (
    <div className="rounded-3xl bg-white p-4">
      <div className="flex items-center justify-between">
        <p className="text-sm text-muted">
          {formatDate(data.blockTimestamp * 1000, false)}
        </p>
      </div>

      <div className="mt-4 grid grid-cols-2 gap-y-4">
        <CaptionItem
          title="Transaction type"
          value={trxType.get(data.transactionType)!}
        />
        <CaptionItem title="Transaction hash" value={shortenWallet(data.hash)} withCopy />
        <CaptionItem
          title="Sender's wallet"
          value={shortenWallet(data.fromAddress)}
          withCopy
        />
        <CaptionItem
          title="Recipient's wallet"
          value={shortenWallet(data.toAddress)}
          withCopy
        />
        <CaptionItem
          title="Amount of stETH"
          value={`${formatNumber(data.ethAmount)} stETH`}
        />
        <CaptionItem
          title="Network fee"
          value={`${formatNumber(data.networkFee)} stETH`}
        />
      </div>
    </div>
  )
}
