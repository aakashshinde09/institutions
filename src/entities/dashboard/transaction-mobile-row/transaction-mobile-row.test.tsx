import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { TransactionMobileRow } from './transaction-mobile-row.component'
import { MantineProvider } from '@mantine/core'

const mock = {
  id: 12,
  hash: '0xd92fe71e6167532f61ccc4a6c5b49acbe02d818840b22736dec66ec4d8dd9549',
  transactionType: '1',
  fromAddress: '0xc946cb236481c159f460b212b34ab246dac37fcd',
  toAddress: '0xa72afa79826df92b88831b9ebd1a459a75345231',
  ethAmount: 0.01,
  networkFee: 0.001197551826812835,
  blockNumber: 1470553,
  blockTimestamp: 1714751892,
}

describe('<TransactionMobileRow/>', () => {
  test('Renders correctly', () => {
    const component = render(<TransactionMobileRow data={mock} />, {wrapper: MantineProvider})

    expect(component).toMatchSnapshot()
  })
})
