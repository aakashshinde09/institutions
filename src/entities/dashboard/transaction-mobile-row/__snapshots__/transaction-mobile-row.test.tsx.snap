// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TransactionMobileRow/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        class="rounded-3xl bg-white p-4"
      >
        <div
          class="flex items-center justify-between"
        >
          <p
            class="text-sm text-muted"
          >
            03/05/2024 18:58
          </p>
        </div>
        <div
          class="mt-4 grid grid-cols-2 gap-y-4"
        >
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Transaction type
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              Stake
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Transaction hash
            </p>
            <span
              class="text-2xl font-bold max-md:text-base flex items-center mt-1 gap-1"
            >
              0xd92...d9549
              <button
                class="mantine-focus-auto h-6 w-6 hover:scale-105 active:scale-95 transition shrink-0 text-primary m_87cf2631 mantine-UnstyledButton-root"
                type="button"
              />
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Sender's wallet
            </p>
            <span
              class="text-2xl font-bold max-md:text-base flex items-center mt-1 gap-1"
            >
              0xc94...37fcd
              <button
                class="mantine-focus-auto h-6 w-6 hover:scale-105 active:scale-95 transition shrink-0 text-primary m_87cf2631 mantine-UnstyledButton-root"
                type="button"
              />
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Recipient's wallet
            </p>
            <span
              class="text-2xl font-bold max-md:text-base flex items-center mt-1 gap-1"
            >
              0xa72...45231
              <button
                class="mantine-focus-auto h-6 w-6 hover:scale-105 active:scale-95 transition shrink-0 text-primary m_87cf2631 mantine-UnstyledButton-root"
                type="button"
              />
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Amount of stETH
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.01 stETH
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Network fee
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.001198 stETH
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      data-portal="true"
    />
    <div
      data-portal="true"
    />
    <div
      data-portal="true"
    />
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      class="rounded-3xl bg-white p-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <p
          class="text-sm text-muted"
        >
          03/05/2024 18:58
        </p>
      </div>
      <div
        class="mt-4 grid grid-cols-2 gap-y-4"
      >
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Transaction type
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            Stake
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Transaction hash
          </p>
          <span
            class="text-2xl font-bold max-md:text-base flex items-center mt-1 gap-1"
          >
            0xd92...d9549
            <button
              class="mantine-focus-auto h-6 w-6 hover:scale-105 active:scale-95 transition shrink-0 text-primary m_87cf2631 mantine-UnstyledButton-root"
              type="button"
            />
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Sender's wallet
          </p>
          <span
            class="text-2xl font-bold max-md:text-base flex items-center mt-1 gap-1"
          >
            0xc94...37fcd
            <button
              class="mantine-focus-auto h-6 w-6 hover:scale-105 active:scale-95 transition shrink-0 text-primary m_87cf2631 mantine-UnstyledButton-root"
              type="button"
            />
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Recipient's wallet
          </p>
          <span
            class="text-2xl font-bold max-md:text-base flex items-center mt-1 gap-1"
          >
            0xa72...45231
            <button
              class="mantine-focus-auto h-6 w-6 hover:scale-105 active:scale-95 transition shrink-0 text-primary m_87cf2631 mantine-UnstyledButton-root"
              type="button"
            />
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Amount of stETH
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.01 stETH
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Network fee
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.001198 stETH
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
